plugin/pkg/scheduler/extender_test.go

```diff
@@ -0,0 +1,302 @@
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package scheduler
+
+import (
+	"fmt"
+	"math/rand"
+	"testing"
+
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/plugin/pkg/scheduler/algorithm"
+	schedulerapi "k8s.io/kubernetes/plugin/pkg/scheduler/api"
+)
+
+type fitPredicate func(pod *api.Pod, node *api.Node) (bool, error)
+type priorityFunc func(pod *api.Pod, nodes *api.NodeList) (*schedulerapi.HostPriorityList, error)
+
+type priorityConfig struct {
+	function priorityFunc
+	weight   int
+}
+
+func errorPredicateExtender(pod *api.Pod, node *api.Node) (bool, error) {
+	return false, fmt.Errorf("Some error")
+}
+
+func falsePredicateExtender(pod *api.Pod, node *api.Node) (bool, error) {
+	return false, nil
+}
+
+func truePredicateExtender(pod *api.Pod, node *api.Node) (bool, error) {
+	return true, nil
+}
+
+func machine1PredicateExtender(pod *api.Pod, node *api.Node) (bool, error) {
+	if node.Name == "machine1" {
+		return true, nil
+	}
+	return false, nil
+}
+
+func machine2PredicateExtender(pod *api.Pod, node *api.Node) (bool, error) {
+	if node.Name == "machine2" {
+		return true, nil
+	}
+	return false, nil
+}
+
+func errorPrioritizerExtender(pod *api.Pod, nodes *api.NodeList) (*schedulerapi.HostPriorityList, error) {
+	return &schedulerapi.HostPriorityList{}, fmt.Errorf("Some error")
+}
+
+func machine1PrioritizerExtender(pod *api.Pod, nodes *api.NodeList) (*schedulerapi.HostPriorityList, error) {
+	result := schedulerapi.HostPriorityList{}
+	for _, node := range nodes.Items {
+		score := 1
+		if node.Name == "machine1" {
+			score = 10
+		}
+		result = append(result, schedulerapi.HostPriority{node.Name, score})
+	}
+	return &result, nil
+}
+
+func machine2PrioritizerExtender(pod *api.Pod, nodes *api.NodeList) (*schedulerapi.HostPriorityList, error) {
+	result := schedulerapi.HostPriorityList{}
+	for _, node := range nodes.Items {
+		score := 1
+		if node.Name == "machine2" {
+			score = 10
+		}
+		result = append(result, schedulerapi.HostPriority{node.Name, score})
+	}
+	return &result, nil
+}
+
+func machine2Prioritizer(_ *api.Pod, podLister algorithm.PodLister, nodeLister algorithm.NodeLister) (schedulerapi.HostPriorityList, error) {
+	nodes, err := nodeLister.List()
+	if err != nil {
+		return []schedulerapi.HostPriority{}, err
+	}
+
+	result := []schedulerapi.HostPriority{}
+	for _, node := range nodes.Items {
+		score := 1
+		if node.Name == "machine2" {
+			score = 10
+		}
+		result = append(result, schedulerapi.HostPriority{node.Name, score})
+	}
+	return result, nil
+}
+
+type FakeExtender struct {
```

Comment:
I dont know if there is a formal convention. I see other instances where a structure and their methods are together (scheduler_test.go)